ctapipe/plotting/charge_resolution.py

```diff
@@ -0,0 +1,360 @@
+import numpy as np
+import pandas as pd
+from matplotlib.ticker import FuncFormatter
+from matplotlib import pyplot as plt
+import os
+from ctapipe.core import Component
+from traitlets import Unicode, Int
+
+
+def sum_errors(array):
+    """
+    Simple sum of squares to combine errors
+
+    Parameters
+    ----------
+    array : ndarray
+
+    Returns
+    -------
+    float
+    """
+    return np.sqrt(np.sum(np.power(array, 2)) / array.size)
+
+
+def bin_dataframe(df, n_bins):
+    """
+    Assign a "bin" column to the dataframe to indicate which bin the true
+    charges belong to.
+
+    Bins are assigned in log space.
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+    n_bins : int
+        Number of bins to allow in range
+
+    Returns
+    -------
+    pd.DataFrame
+    """
+    true = df['true'].values
+    min_ = true.min()
+    max_ = (true.max() // 500 + 1) * 500
+    bins = np.geomspace(min_, max_, n_bins)
+    bins = np.append(bins, 10**(np.log10(bins[-1]) +
+                                np.diff(np.log10(bins))[0]))
+    df['bin'] = np.digitize(true, bins, right=True) - 1
+
+    return df
+
+
+class ChargeResolutionPlotter(Component):
+
+    output_path = Unicode(
+        '', help='Output path to save the plot.'
+    ).tag(config=True)
+
+    n_bins = Int(
+        40,
+        help='Number of bins for collecting true charges and combining '
+             'their resolution'
+    ).tag(config=True)
+
+    def __init__(self, config=None, tool=None, **kwargs):
+        """
+        Plots the charge resolution HDF5 file obtained from
+        `ctapipe.analysis.camera.charge_resolution`.
+
+        Also contains the equation from which the charge resolution
+        requirement is defined, which can be plotted alongside the charge
+        resolution curve.
+
+        `ctapipe.tools.plot_charge_resolution` demonstrated the use of this
+        component.
+
+        Parameters
+        ----------
+        config : traitlets.loader.Config
+            Configuration specified by config file or cmdline arguments.
+            Used to set traitlet values.
+            Set to None if no configuration to pass.
+        tool : ctapipe.core.Tool
+            Tool executable that is calling this component.
+            Passes the correct logger to the component.
+            Set to None if no Tool to pass.
+        kwargs
+        """
+        super().__init__(config=config, parent=tool, **kwargs)
+        self._df_pixel = None
+        self._df_camera = None
+
+        self.fig = plt.figure(figsize=(12, 7.42))
+        self.ax = self.fig.add_subplot(111)
+
+        self.ax.set_xlabel("True Charge (p.e.)")
+        self.ax.set_ylabel(
+            r"Fractional Charge Resolution $\frac{{\sigma_Q}}{{Q}}$"
+        )
+
+        if not self.output_path:
+            raise ValueError("Output path must be specified")
+
+    def _set_file(self, path):
```

Comment:
Could this method be renamed to reflect its purpose a little bit clearer? It seems this method opens a file, reads some contents from the file and stores these contents as instance variables for later use.
Maybe something like "read_charge_resolution_from_hdfstore()" or so?

Reply:
I don't think the long function name is necessary, but I will add more to the description. Its a private method that the user never needs to call.

Reply:
I was more convert about your fellow devs than a "user"

Reply:
The example is deliberately long .. I wanted to express what I mean by "a little bit clearer", by using a example. Not saying that this example is necessarily the best choice.